contracts/pcv/PCVGuardian.sol

```diff
@@ -11,54 +11,72 @@ contract PCVGuardian is IPCVGuardian, CoreRef {
+    // ---------- Governor-or-Admin-Only State-Changing API ----------
 
     /// @notice governor-only method to set an address as "safe" to withdraw funds to
-    /// @param anAddress the address to set as safe
-    function setSafeAddress(address anAddress) external override onlyGovernor() {
-        safeAddresses.add(anAddress);
-        emit SafeAddressAdded(anAddress);
+    /// @param pcvDeposit the address to set as safe
+    function setSafeAddress(address pcvDeposit) external override onlyGovernorOrAdmin() {
+        _setSafeAddress(pcvDeposit);
     }
 
-    // ---------- Governor-or-Guardian-Only State-Changing API ----------
+    /// @notice batch version of setSafeAddress
+    /// @param safeAddresses the addresses to set as safe, as calldata
+    function setSafeAddresses(address[] calldata safeAddresses) external override onlyGovernorOrAdmin() {
+        for(uint256 i=0; i<safeAddresses.length; i++) {
+            _setSafeAddress(safeAddresses[i]);
+        }
+    }
+
+    // ---------- Governor-or-Admin-Or-Guardian-Only State-Changing API ----------
 
     /// @notice governor-or-guardian-only method to un-set an address as "safe" to withdraw funds to
-    /// @param anAddress the address to un-set as safe
-    function unsetSafeAddress(address anAddress) external override onlyGuardianOrGovernor() {
-        safeAddresses.remove(anAddress);
-        emit SafeAddressRemoved(anAddress);
+    /// @param pcvDeposit the address to un-set as safe
+    function unsetSafeAddress(address pcvDeposit) external override isGovernorOrGuardianOrAdmin() {
+        _unsetSafeAddress(pcvDeposit);
+    }
+
+    /// @notice batch version of unsetSafeAddresses
+    /// @param safeAddresses the addresses to un-set as safe
+    function unsetSafeAddresses(address[] calldata safeAddresses) external override isGovernorOrGuardianOrAdmin() {
+        for(uint256 i=0; i<safeAddresses.length; i++) {
+            _unsetSafeAddress(safeAddresses[i]);
+        }
     }
 
     /// @notice governor-or-guardian-only method to withdraw funds from a pcv deposit, by calling the withdraw() method on it
     /// @param pcvDeposit the address of the pcv deposit contract
     /// @param safeAddress the destination address to withdraw to
     /// @param amount the amount to withdraw
-    /// @param unpauseBefore if true, the pcv contract will be unpaused before the withdraw
     /// @param pauseAfter if true, the pcv contract will be paused after the withdraw
-    function withdrawToSafeAddress(address pcvDeposit, address safeAddress, uint256 amount, bool unpauseBefore, bool pauseAfter) external override onlyGuardianOrGovernor() {
+    function withdrawToSafeAddress(address pcvDeposit, address safeAddress, uint256 amount, bool pauseAfter) external override isGovernorOrGuardianOrAdmin() {
         require(isSafeAddress(safeAddress), "Provided address is not a safe address!");
 
-        if (unpauseBefore) {
+        if (CoreRef(pcvDeposit).paused()) {
```

Comment:
Nit: Prefer some kind of IPausable interface to casting as CoreRef here.

Reply:
we can use Pausable (which is abstract anyways) for checking pause state, but pause() and unpause() themselves are defined in ICoreRef - doesn't makes sense to me to redefine them in another interface

Reply:
updated with Pausable where I could